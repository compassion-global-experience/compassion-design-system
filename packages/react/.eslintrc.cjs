module.exports = {
  extends: [
    'standard-jsx',
    'standard-react',
    'plugin:jsx-a11y/recommended',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
    'plugin:storybook/recommended',
  ],
  env: {
    browser: true,
    es6: true,
    jest: true,
    node: false,
  },
  rules: {
    'jsx-quotes': ['error', 'prefer-double'],
  },
};
