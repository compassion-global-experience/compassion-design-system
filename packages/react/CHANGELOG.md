# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2023-04-08)

### Features

- **react:** css usage and documentation ([11e317c](https://github.com/compassion-global-experience/compassion-design-system/commit/11e317cca4ecdaa385e86d3a5571d7dca554685d))
- **react:** re-export css package ([67549a5](https://github.com/compassion-global-experience/compassion-design-system/commit/67549a58de3de6e02b58c5bdb6f4605001aae783))
- **react:** switch between themes in storybook ([79119ba](https://github.com/compassion-global-experience/compassion-design-system/commit/79119ba48ed3410eaebe9a91ccd076b05c2580c1))
