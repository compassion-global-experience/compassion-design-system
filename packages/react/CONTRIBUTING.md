# Contributing to the `react-web` Package

The `react` package is structured with a component-per-directory convention
inside the `src` directory. Each component directory generally contains 3 items:

- A **TSX file** that describes the component‚Äôs structure and behavior
- A **.stories.ts file** that illustrates most of the ways the component can
  be configured
- An **index.ts** file that exports the TSX component‚Äôs contents

## Workflow

The package uses [Storybook](https://storybook.js.org/docs/react/get-started/introduction)
to facilitate component development and documentation. Storybook uses the files
mentioned above to render components in their various states, and serves as
documentation of those components by surfacing code comments, `propTypes`, and
`defaultProps` values.

To launch Storybook, make sure you‚Äôre in the [/packages/react](/packages/react)
directory and run the command `npm run storybook`. After the project is built,
Storybook will launch in your default browser. New components or modifications
will be displayed in real time! üéâ

The configurations outlined in each component‚Äôs .stories.ts file can be seen in
Storybook‚Äôs **Docs** tab.

## Commit Messages & Versioning

Every package in the compassion-gds monorepo adheres to
[semantic versioning](https://semver.org/). In order to simplify the versioning
and release process, please be sure your commit messages adhere to the
[‚ÄúConventional Commits‚Äù pattern](https://www.conventionalcommits.org/en/v1.0.0/#summary).
You may wish to install the [commitizen/cz-cli](https://github.com/commitizen/cz-cli)
to make this near-bulletproof.

## Making a Pull Request

Once your changes are ready, open a pull request against the `development`
branch of the project, and it will be merged in as quickly as possible!
