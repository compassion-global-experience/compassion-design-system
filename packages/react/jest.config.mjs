/*
 * For a detailed explanation regarding each configuration property, visit:
 * https://jestjs.io/docs/configuration
 */

export default {
  // A preset that is used as a base for Jest's configuration
  preset: 'ts-jest/presets/default-esm',

  // Automatically clear mock calls, instances, contexts and results before every test
  clearMocks: true,

  // Indicates whether the coverage information should be collected while executing the test
  collectCoverage: true,

  // The directory where Jest should output its coverage files
  coverageDirectory: "coverage",

  // Indicates which provider should be used to instrument code for coverage
  coverageProvider: "v8",

  // The test environment that will be used for testing
  testEnvironment: "jsdom",

  // A map from regular expressions to module names or to arrays of module names that allow to stub out resources with a single module
  moduleNameMapper: {
    ".(css|less|scss)$": "identity-obj-proxy",
  },

  // Whether to use watchman for file crawling
  watchman: true,
};
