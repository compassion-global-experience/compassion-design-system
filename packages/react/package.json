{
  "name": "@compassion-gds/react",
  "version": "0.2.0-alpha.2",
  "description": "",
  "type": "module",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "files": [
    "dist"
  ],
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "browser": {
        "default": "./dist/esm/index.js"
      },
      "import": "./dist/esm/index.js",
      "require": "./dist/cjs/index.js"
    },
    "./css/base.css": "./dist/css/reset.css",
    "./css/light.css": "./dist/css/cds-light.css",
    "./css/dark.css": "./dist/css/cds-dark.css",
    "./css/*": "./dist/css/*"
  },
  "scripts": {
    "dev": "rollup -c -w",
    "copy-css": "bash ./scripts/copy-css.sh",
    "prepare": "rollup -c && npm run copy-css",
    "test": "jest",
    "storybook": "storybook dev -p 6007",
    "build:storybook": "storybook build --quiet",
    "percy": "cross-env PERCY_TOKEN=$PERCY_TOKEN_REACT percy storybook ./storybook-static",
    "percy-snapshots": "npm run build:storybook && npm run percy"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/compassion-global-experience/compassion-design-system.git",
    "directory": "packages/react"
  },
  "keywords": [
    "compassion",
    "compassion-gds",
    "gds",
    "react"
  ],
  "author": "Compassion XD",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/compassion-global-experience/compassion-design-system/issues"
  },
  "homepage": "https://github.com/compassion-global-experience/compassion-design-system#readme",
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@compassion-gds/core": "^0.1.0-alpha.0",
    "@rollup/plugin-commonjs": "^24.0.1",
    "@rollup/plugin-node-resolve": "^15.0.1",
    "@rollup/plugin-replace": "^5.0.2",
    "@rollup/plugin-typescript": "^11.1.0",
    "@storybook/addon-actions": "^7.0.2",
    "@storybook/addon-docs": "^7.0.2",
    "@storybook/addon-essentials": "^7.0.2",
    "@storybook/addon-interactions": "^7.0.2",
    "@storybook/addon-links": "^7.0.2",
    "@storybook/preset-react-webpack": "7.0.2",
    "@storybook/react": "7.0.2",
    "@storybook/react-webpack5": "7.0.2",
    "@storybook/testing-library": "^0.0.14-next.1",
    "@testing-library/react": "^13.4.0",
    "@types/jest": "^29.4.0",
    "@types/react": "^18.0.27",
    "babel-loader": "^8.3.0",
    "eslint": "^8.33.0",
    "eslint-config-standard-jsx": "^11.0.0",
    "eslint-config-standard-react": "^13.0.0",
    "eslint-plugin-jsx-a11y": "^6.7.1",
    "eslint-plugin-react": "^7.32.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-storybook": "^0.6.10",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "phosphor-react": "^1.4.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^3.20.2",
    "rollup-plugin-dts": "^5.1.1",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.2",
    "storybook": "7.0.2",
    "ts-jest": "^29.0.5",
    "webpack": "^5.75.0"
  },
  "peerDependencies": {
    "react": "16.8.x || 17.x || 18.x",
    "react-dom": "16.8.x || 17.x || 18.x"
  }
}
