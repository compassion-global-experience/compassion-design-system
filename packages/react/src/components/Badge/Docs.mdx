import { Markdown } from '@storybook/blocks';
import {
  Canvas,
  Controls,
  Meta,
  Source,
  Stories,
  Story,
  Title,
} from '@storybook/addon-docs';

import * as BadgeStories from './Badge.stories.tsx';

import _01_Overview from '@compassion-design-system/core/src/components/Badge/docs/01_Overview.md';
import _02_Usage from '@compassion-design-system/core/src/components/Badge/docs/02_Usage.md';
import _03_Status from '@compassion-design-system/core/src/components/Badge/docs/03_Status.md';
import _04_BorderRadius from '@compassion-design-system/core/src/components/Badge/docs/04_BorderRadius.md';
import _05_OtherProperties from '@compassion-design-system/core/src/components/Badge/docs/05_OtherProperties.md';
import _06_BestPractices from '@compassion-design-system/core/src/components/Badge/docs/06_BestPractices.md';

<Meta of={BadgeStories} />

<Title />

<Markdown>{_01_Overview}</Markdown>

<Canvas sourceState="shown" withToolbar="true" />
<Controls />

<Markdown>{_02_Usage}</Markdown>

## Component Properties

<Markdown>{_03_Status}</Markdown>
<Canvas of={BadgeStories.Status} />

<Markdown>{_04_BorderRadius}</Markdown>
<Canvas of={BadgeStories.BorderRadius} />

<Markdown>{_05_OtherProperties}</Markdown>
<Canvas of={BadgeStories.WithIcon} />

<Markdown>{_06_BestPractices}</Markdown>
