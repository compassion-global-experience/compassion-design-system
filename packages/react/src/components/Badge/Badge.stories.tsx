import { Meta } from '@storybook/react';
import Badge from './Badge';
import { cdsIconMap } from '../Icon/Icon';

const meta: Meta<typeof Badge> = {
  title: 'Components/Badge',
  component: Badge,
};

export default meta;

export const Default = {
  argTypes: {
    label: {
      control: { type: 'text' },
    },
    showLabel: {
      control: { type: 'boolean' },
    },
    status: {
      control: { type: 'select' },
    },
    borderRadius: {
      control: { type: 'select' },
    },
    icon: {
      control: { type: 'select' },
      options: Object.keys(cdsIconMap).sort(),
    },
    iconPosition: {
      control: { type: 'select' },
    },
  },
  args: {
    label: 'Badge',
  },
};

const Template = (args) => {
  return <Badge {...args} />;
};

export const Status = (args) => (
  <>
    <Badge label="Primary" status="primary" {...args} />
    <Badge label="Info" status="info" {...args} />
    <Badge label="Success" status="success" {...args} />
    <Badge label="Warning" status="warning" {...args} />
    <Badge label="Danger" status="danger" {...args} />
    <Badge label="Neutral" {...args} />
  </>
);

export const BorderRadius = (args) => (
  <>
    <Badge label="Rounded Badge" {...args} />
    <Badge label="Pill Badge" borderRadius="pill" {...args} />
  </>
);

export const WithIcon = (args) => (
  <>
    <Badge
      label="With iconPosition: end"
      icon="ph-arrow-right"
      iconPosition="end"
      {...args}
    />
    <Badge label="With iconPosition: start" icon="ph-warning" {...args} />
    <Badge label="" icon="ph-flag" {...args} />
  </>
);
