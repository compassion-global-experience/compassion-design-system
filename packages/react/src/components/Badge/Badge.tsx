import { ReactNode } from 'react';
import styles from '@compassion-design-system/core/src/components/Badge/badge.module.css';
import { getClasses } from '../../utils/classes';
import Icon, { cdsIconMap } from '../Icon/Icon';

export interface BadgeProps {
  status?: 'primary' | 'warning' | 'danger' | 'success' | 'info' | 'neutral';
  label: string;
  borderRadius?: 'rounded' | 'pill';
  icon?: keyof typeof cdsIconMap;
  iconPosition?: 'start' | 'end';
  showLabel?: boolean;
}

const Badge = ({
  label,
  status = 'neutral',
  borderRadius = 'rounded',
  icon,
  iconPosition = 'start',
  showLabel = true,
}: BadgeProps) => {
  const badgeClasses = getClasses(styles, [
    'cds-badge',
    `cds-badge__${status}`,
    `cds-badge__${borderRadius}`,
    icon ? `cds-badge__icon-${iconPosition}` : '',
  ]).trim();

  return (
    <div className={badgeClasses}>
      {showLabel && <span>{label}</span>}
      {icon && (
        <Icon icon={icon} description={`Icon for badge: ${label}`} size="xs" />
      )}
    </div>
  );
};

export default Badge;
