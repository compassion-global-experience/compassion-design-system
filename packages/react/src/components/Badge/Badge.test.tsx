import { render } from '@testing-library/react';
import Badge from './Badge';

describe('Badge', () => {
  it('should render default props correctly', () => {
    const { container } = render(<Badge label="Test Badge" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a badge with an icon', () => {
    const { container } = render(<Badge label="Test Badge" icon="church" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a badge with an icon at the end', () => {
    const { container } = render(
      <Badge label="Test Badge" icon="church" iconPosition="end" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a badge with a pill border radius', () => {
    const { container } = render(
      <Badge label="Test Badge" borderRadius="pill" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an icon only badge', () => {
    const { container } = render(
      <Badge label="Test Badge" icon="church" showLabel={false} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
