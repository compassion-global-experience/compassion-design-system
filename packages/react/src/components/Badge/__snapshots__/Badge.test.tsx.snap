// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render a badge with a pill border radius 1`] = `
<div
  class="cds-badge cds-badge__neutral cds-badge__pill"
>
  <span>
    Test Badge
  </span>
</div>
`;

exports[`Badge should render a badge with an icon 1`] = `
<div
  class="cds-badge cds-badge__neutral cds-badge__rounded cds-badge__icon-start"
>
  <span>
    Test Badge
  </span>
  <svg
    class="cds-icon__svg--default"
    fill="currentColor"
    height="12"
    viewBox="0 0 256 256"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Icon for badge: Test Badge
    </title>
    <path
      d="M228.12,145.14,192,123.47V104a8,8,0,0,0-4-7L136,67.36V48h16a8,8,0,0,0,0-16H136V16a8,8,0,0,0-16,0V32H104a8,8,0,0,0,0,16h16V67.36L68,97.05a8,8,0,0,0-4,7v19.47L27.88,145.14A8,8,0,0,0,24,152v64a8,8,0,0,0,8,8h80a8,8,0,0,0,8-8V168a8,8,0,0,1,16,0v48a8,8,0,0,0,8,8h80a8,8,0,0,0,8-8V152A8,8,0,0,0,228.12,145.14ZM40,156.53l24-14.4V208H40ZM128,144a24,24,0,0,0-24,24v40H80V108.64l48-27.43,48,27.43V208H152V168A24,24,0,0,0,128,144Zm88,64H192V142.13l24,14.4Z"
    />
  </svg>
</div>
`;

exports[`Badge should render a badge with an icon at the end 1`] = `
<div
  class="cds-badge cds-badge__neutral cds-badge__rounded cds-badge__icon-end"
>
  <span>
    Test Badge
  </span>
  <svg
    class="cds-icon__svg--default"
    fill="currentColor"
    height="12"
    viewBox="0 0 256 256"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Icon for badge: Test Badge
    </title>
    <path
      d="M228.12,145.14,192,123.47V104a8,8,0,0,0-4-7L136,67.36V48h16a8,8,0,0,0,0-16H136V16a8,8,0,0,0-16,0V32H104a8,8,0,0,0,0,16h16V67.36L68,97.05a8,8,0,0,0-4,7v19.47L27.88,145.14A8,8,0,0,0,24,152v64a8,8,0,0,0,8,8h80a8,8,0,0,0,8-8V168a8,8,0,0,1,16,0v48a8,8,0,0,0,8,8h80a8,8,0,0,0,8-8V152A8,8,0,0,0,228.12,145.14ZM40,156.53l24-14.4V208H40ZM128,144a24,24,0,0,0-24,24v40H80V108.64l48-27.43,48,27.43V208H152V168A24,24,0,0,0,128,144Zm88,64H192V142.13l24,14.4Z"
    />
  </svg>
</div>
`;

exports[`Badge should render an icon only badge 1`] = `
<div
  class="cds-badge cds-badge__neutral cds-badge__rounded cds-badge__icon-start"
>
  <svg
    class="cds-icon__svg--default"
    fill="currentColor"
    height="12"
    viewBox="0 0 256 256"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Icon for badge: Test Badge
    </title>
    <path
      d="M228.12,145.14,192,123.47V104a8,8,0,0,0-4-7L136,67.36V48h16a8,8,0,0,0,0-16H136V16a8,8,0,0,0-16,0V32H104a8,8,0,0,0,0,16h16V67.36L68,97.05a8,8,0,0,0-4,7v19.47L27.88,145.14A8,8,0,0,0,24,152v64a8,8,0,0,0,8,8h80a8,8,0,0,0,8-8V168a8,8,0,0,1,16,0v48a8,8,0,0,0,8,8h80a8,8,0,0,0,8-8V152A8,8,0,0,0,228.12,145.14ZM40,156.53l24-14.4V208H40ZM128,144a24,24,0,0,0-24,24v40H80V108.64l48-27.43,48,27.43V208H152V168A24,24,0,0,0,128,144Zm88,64H192V142.13l24,14.4Z"
    />
  </svg>
</div>
`;

exports[`Badge should render default props correctly 1`] = `
<div
  class="cds-badge cds-badge__neutral cds-badge__rounded"
>
  <span>
    Test Badge
  </span>
</div>
`;
